components/brave_rewards/browser/rewards_p3a.cc

```diff
@@ -182,61 +182,57 @@ double CalcWalletBalance(base::flat_map<std::string, double> wallets,
   return balance_minus_grant;
 }
 
-void RecordRewardsEnabledDuration(PrefService* prefs, bool rewards_enabled) {
-  base::Time enabled_timestamp = prefs->GetTime(prefs::kEnabledTimestamp);
-  auto enabled_duration = RewardsEnabledDuration::kNever;
+void RecordAdsEnabledDuration(PrefService* prefs, bool ads_enabled) {
+  base::Time enabled_timestamp = prefs->GetTime(prefs::kAdsEnabledTimestamp);
+  base::TimeDelta enabled_time_delta =
+      prefs->GetTimeDelta(prefs::kAdsEnabledTimeDelta);
+  AdsEnabledDuration enabled_duration;
 
   if (enabled_timestamp.is_null()) {
     // No previous timestamp, so record one of the non-duration states.
-    if (!rewards_enabled) {
-      // Rewards have been disabled with no previous timestamp.
-      // Probably they've been on since we started measuring.
-      // Ignore this interval since we can't measure it and treat it
-      // the same as never having enabled.
-      enabled_duration = RewardsEnabledDuration::kNever;
-    } else {
-      // Rewards have been enabled.
+    if (ads_enabled) {
+      // Ads have been enabled.
       // Remember when so we can measure the duration on later changes.
-      prefs->SetTime(prefs::kEnabledTimestamp, base::Time::Now());
-      enabled_duration = RewardsEnabledDuration::kStillEnabled;
+      prefs->SetTime(prefs::kAdsEnabledTimestamp, base::Time::Now());
+      enabled_duration = AdsEnabledDuration::kStillEnabled;
     }
   } else {
     // Previous timestamp available.
-    if (!rewards_enabled) {
-      // Rewards have been disabled. Record the duration they were on.
-      // Set the threshold at three units so each bin represents the
-      // nominal value as an order-of-magnitude: more than three days
-      // is a week, more than three weeks is a month, and so on.
-      constexpr int threshold = 3;
-      constexpr int days_per_week = 7;
-      constexpr double days_per_month = 30.44;  // average length
-      base::TimeDelta duration = base::Time::Now() - enabled_timestamp;
-      VLOG(1) << "Rewards disabled after " << duration;
-      if (duration < base::Hours(threshold)) {
-        enabled_duration = RewardsEnabledDuration::kHours;
-      } else if (duration < base::Days(threshold)) {
-        enabled_duration = RewardsEnabledDuration::kDays;
-      } else if (duration < base::Days(threshold * days_per_week)) {
-        enabled_duration = RewardsEnabledDuration::kWeeks;
-      } else if (duration < base::Days(threshold * days_per_month)) {
-        enabled_duration = RewardsEnabledDuration::kMonths;
-      } else {
-        enabled_duration = RewardsEnabledDuration::kQuarters;
-      }
+    if (!ads_enabled) {
+      // Ads have been disabled. Record the duration they were on.
+      enabled_time_delta = base::Time::Now() - enabled_timestamp;
```

Comment:
What happens if `enabled_timestamp` is the default value here? Seems like that would report `kQuarters` for anyone whose profile had ads enabled since before the pref value was added. Maybe that's ok, but should probably be commented.

Reply:
This block of code would not be executed if `enabled_timestamp` was set to null (the default). The `enabled_time_delta` would be zero, and `kNever` would be reported.

Reply:
Ok, I see. It's protected by the `if (enabled_timestamp.in_null())` check above. Sorry, I read the diff incorrectly.